### ISSUE TITLE
[React] - baseUrl ignored when using next-auth in a chrome extension

### ISSUE THREAD

**Issue by jonluca** [nextauthjs/next-auth#7012]
### Environment

  System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 777.19 MB / 64.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.8.1 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 9.5.1 - /opt/homebrew/bin/npm
  Browsers:
    Brave Browser: 111.1.49.120
    Chrome: 111.0.5563.64
    Firefox: 111.0
    Safari: 16.3

### Reproduction URL

NA

### Describe the issue

Use a `SessionProvider` on a domain different from the one which the next-auth app is running - for example, in a chrome extension.

### How to reproduce

In a chrome extension, use the following code:

```ts
export async function fetchSession() {
  const res = await fetch(config.ui + "/api/auth/session", { mode: "cors" });
  const session = await res.json();
  if (Object.keys(session).length) {
    return session;
  }
  return null;
}
const onLogin = async (path: string) => {
  const url = new URL(config.ui + path);
  await browser.tabs.create({ url: url.toString(), active: true });
};
export function useSessionData() {
  const { data } = useQuery<Session>(["session"], fetchSession, {
    onSettled(data) {
      if (data) {
        return;
      }
      onLogin("/api/auth/signin?error=SessionExpired");
    },
  });
  return data;
}

const InjectSession = ({ children }: React.PropsWithChildren) => {
  const session = useSessionData();
  return (
    <SessionProvider baseUrl={config.ui} session={session}>
      {children}
    </SessionProvider>
  );
};

```

And view that on the page, a `/api/auth/session` request is initiated to the current origin, not the baseUrl.

You can get around this by monkey patching fetch before starting react, but that doesn't feel like a safe workaround.

### Expected behavior

When setting the `baseUrl` prop on a `SessionProvider`, I'd expect all `/api/auth/*` endpoints to go to that domain. However, in react, it'll initiate a fetch request on the client to just `/api/auth/<path>`, which for a chrome extension will be incorrect

This can be seen on L38 of https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/client/_utils.ts#L38

```
export function apiBaseUrl(__NEXTAUTH: AuthClientConfig) {
  if (typeof window === "undefined") {
    // Return absolute path when called server side
    return `${__NEXTAUTH.baseUrlServer}${__NEXTAUTH.basePathServer}`
  }
  // Return relative path when called client side
  return __NEXTAUTH.basePath
}

```

`apiBaseUrl` only returns the basePath, not the baseUrl + basePath

**Comment by jonluca** [comment]
The monkey patch for the chrome extension looks like this:

```tsx
const monkeyPatchFetchForNextAuth = () => {
  const originalFetch = window.fetch;
  window.fetch = async (url, options) => {
    if (typeof url === "string" && url.startsWith("/api/auth")) {
      url = config.ui + url;
    }
    return originalFetch(url, options);
  };
};
```

**Comment by balazsorban44** [comment]
Thanks, but technically `next-auth` + Chrome Extension is out of scope.

We made it work without React by hardcoding the path to the session endpoint in this example https://github.com/domeccleston/sharegpt/blob/e26d2617312f01e4813880898671526514211549/sharegpt-extension/popup.js#L8

Your workaround should be fine. The `next-auth` client was designed to be on the same URL as the backend, so it has never come up as an issue since browsers allow relative URLs via `fetch` to the same origin.
